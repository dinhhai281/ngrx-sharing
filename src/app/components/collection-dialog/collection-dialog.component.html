<h1 mat-dialog-title class="text-lg text-pink-700">
  Create New Collection
</h1>
<form mat-dialog-content (ngSubmit)="submit$.next()" [formGroup]="form">
  <input
    class="p-2 flex-grow rounded text-pink-800 focus:outline-none border-2 border-pink-700"
    placeholder="Collection Name"
    formControlName="collectionName"
  />
</form>
<div mat-dialog-actions align="end">
  <button
    mat-button
    color="primary"
    class="focus:outline-none"
    (click)="submit$.next()"
    >
      Create
    </button>
</div>
