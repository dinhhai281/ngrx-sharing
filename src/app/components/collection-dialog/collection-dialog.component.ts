import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/core';
import { Validators } from '@angular/forms';
import { FormBuilder } from '@ngneat/reactive-forms';
import { Subject } from 'rxjs';
import { map, take, withLatestFrom } from 'rxjs/operators';

@Component({
  selector: 'app-collection-dialog',
  templateUrl: './collection-dialog.component.html',
  styleUrls: ['./collection-dialog.component.scss'],
  encapsulation: ViewEncapsulation.None,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CollectionDialogComponent {
  submit$ = new Subject<void>();
  form = this.fb.group<{ collectionName: string }>({
    collectionName: this.fb.control('', Validators.required),
  });

  constructor(private readonly fb: FormBuilder) {}

  get value$() {
    return this.submit$.pipe(
      withLatestFrom(this.form.value$),
      map(([, fromValue]) => fromValue.collectionName),
      take(1),
    );
  }
}
