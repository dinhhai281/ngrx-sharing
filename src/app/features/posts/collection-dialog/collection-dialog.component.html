<h1 mat-dialog-title class="text-lg text-pink-700">
  Create New Collection
</h1>
<div mat-dialog-content>
  <input
    class="p-2 flex-grow rounded text-pink-800 focus:outline-none border-2 border-pink-700"
    placeholder="Collection Name"
    [formControl]="collectionName"
  />
</div>
<div mat-dialog-actions align="end">
  <button
    mat-button
    color="primary"
    class="focus:outline-none"
    (click)="submit$.next()"
    >
      Create
    </button>
</div>
