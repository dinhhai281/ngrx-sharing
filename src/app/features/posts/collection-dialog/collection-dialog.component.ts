import { ChangeDetectionStrategy, Component, OnInit, ViewEncapsulation } from '@angular/core';
import { Validators } from '@angular/forms';
import { FormControl } from '@ngneat/reactive-forms';
import { Subject } from 'rxjs';
import { map, take, withLatestFrom } from 'rxjs/operators';

@Component({
  selector: 'app-collection-dialog',
  templateUrl: './collection-dialog.component.html',
  styleUrls: ['./collection-dialog.component.scss'],
  encapsulation: ViewEncapsulation.None,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CollectionDialogComponent implements OnInit {
  collectionName = new FormControl<string>('', Validators.required);
  submit$ = new Subject<void>();

  constructor() {}

  get value$() {
    return this.submit$.pipe(
      withLatestFrom(this.collectionName.value$),
      map(([, collectionName]) => collectionName),
      take(1),
    );
  }

  ngOnInit() {}
}
