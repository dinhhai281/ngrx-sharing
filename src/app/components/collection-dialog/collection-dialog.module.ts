import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { MatDialogModule } from '@angular/material/dialog';

import { CollectionDialogComponent } from './collection-dialog.component';

@NgModule({
  declarations: [CollectionDialogComponent],
  imports: [CommonModule, MatDialogModule, ReactiveFormsModule],
  exports: [CollectionDialogComponent],
})
export class CollectionDialogModule {}
