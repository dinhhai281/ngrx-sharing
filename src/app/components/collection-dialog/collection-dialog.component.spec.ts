import { ReactiveFormsModule } from '@angular/forms';
import { MatDialogModule } from '@angular/material/dialog';
import { render, RenderResult } from '@testing-library/angular';

import { CollectionDialogComponent } from './collection-dialog.component';

describe('CollectionDialogComponent', () => {
  let component: RenderResult<CollectionDialogComponent>;

  beforeEach(async () => {
    component = await render(CollectionDialogComponent, {
      imports: [MatDialogModule, ReactiveFormsModule],
    });
  });

  it('should match snapshot', () => {
    expect(component.container).toMatchSnapshot();
  });
});
