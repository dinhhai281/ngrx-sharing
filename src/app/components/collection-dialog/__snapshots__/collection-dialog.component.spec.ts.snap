// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionDialogComponent should match snapshot 1`] = `
<div
  id="root0"
>
  <h1
    class="text-lg text-pink-700 mat-dialog-title"
    mat-dialog-title=""
  >
     Create New Collection

  </h1>
  <form
    class="mat-dialog-content ng-untouched ng-pristine ng-valid"
    mat-dialog-content=""
    novalidate=""
  >
    <input
      class="p-2 flex-grow rounded text-pink-800 focus:outline-none border-2 border-pink-700 ng-untouched ng-pristine ng-valid"
      formcontrolname="collectionName"
      placeholder="Collection Name"
    />
  </form>
  <div
    align="end"
    class="mat-dialog-actions"
    mat-dialog-actions=""
  >
    <button
      class="focus:outline-none"
      color="primary"
      mat-button=""
    >
       Create 
    </button>
  </div>
</div>
`;
